templates/crossplane/apis/crd.go.tpl

```diff
@@ -21,15 +21,29 @@ type {{ .CRD.Kind }}Parameters struct {
 	// Region is which region the {{ .CRD.Kind }} will be created.
 	// +kubebuilder:validation:Required
 	Region string `json:"region"`
-	{{- range $fieldName, $field := .CRD.SpecFields }}
-	{{- if $field.ShapeRef }}
+{{- range $fieldName, $field := .CRD.SpecFields }}
+	{{ if $field.ShapeRef }}
 	{{ $field.ShapeRef.Documentation }}
 	{{- end }}
-	{{ if $field.IsRequired }} // +kubebuilder:validation:Required
+	{{- if $field.ReferencedType}}
+	{{ $field.Names.Camel }} {{ $field.GoType }} `json:"{{ $field.Names.CamelLower }},omitempty"`
+
+	// {{ $field.Names.Camel }}Ref is a reference to an {{ $field.ReferencedType }} used
+	// to set the {{ $field.Names.Camel }} field.
+	// +optional
+	{{ $field.Names.Camel }}Ref {{if Contains $field.GoType "[]" -}}[]xpv1.Reference{{- else -}}*xpv1.Reference{{- end -}} `json:"{{ $field.Names.CamelLower }}Ref,omitempty"`
+
+	// {{ $field.Names.Camel }}Selector selects references to {{ $field.ReferencedType }}
+	// used to set the {{ $field.Names.Camel }}.
+  // +optional
```

Comment:
indent :)

Reply:
Ugh it's Goland again, thanks for the heads-up. Fixed it.